File: stack/javascript/stack.js
// Implenet Stack Datastructure
// Stack from scratch
class Stack {
    stack = [];
    limit = 5;
    
    // Push
    push(element) {
        !this.isFull() && this.stack.push(element); // condition and expression
    }
    
    // Pop
    pop() {
        !this.isEmpty() && this.stack.pop();
    }
    
    // IsEmpty
    isEmpty() {
        return this.stack.length === 0;
    }
    
    // IsFull
    isFull() {
        return this.stack.length === this.limit;
    }
    
    // Peek or Top
    peek() {
        return !this.isEmpty() && this.stack[this.stack.length - 1];
    }

    // display current stack data
    logStack() {
        return this.stack;
    }
}

// Usage
const s = new Stack();

const old_stack = s.logStack();

console.log('old_stack', old_stack);

s.push(1);
s.push(2);
s.push(3);
s.push(4);
s.push(5);

console.log("peek: ", s.peek());

const current_stack = s.logStack();

console.log('current_stack', current_stack);

File: stack/python/Stack.py
class Stack:
    def __init__(self, limit=3):  # Initialize stack with a size limit
        self.stack = []
        self.limit = limit
    
    def push(self, element):
        """Add an element to the stack if it's not full."""
        if not self.isFull():
            self.stack.append(element)
        else:
            print("Stack is full. Cannot push.")
    
    def pop(self):
        """Remove and return the top element if the stack is not empty."""
        if not self.isEmpty():
            return self.stack.pop()
        print("Stack is empty. Cannot pop.")
    
    def isEmpty(self):
        """Check if the stack is empty."""
        return len(self.stack) == 0
    
    def isFull(self):
        """Check if the stack is full."""
        return len(self.stack) == self.limit
    
    def top(self):
        """Show the top element without removing it."""
        if not self.isEmpty():
            print(self.stack[-1])
        else:
            print("Stack is empty.")
    
    def display(self):
        """Display all elements in the stack."""
        print(self.stack)

# Example usage
stack = Stack()
stack.push(1)
stack.push(2)
stack.push(3)
stack.push(4)  # This will show "Stack is full"
stack.display()
stack.pop()
stack.top()
stack.display()